src/Performance/Memory/MemoryUsageProvider.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Performance\Memory;
+
+/**
+ * @internal
+ */
+final class MemoryUsageProvider
+{
+    public function getPeakUsage(): int
+    {
+        return memory_get_peak_usage(true);
```

Comment:
For me this is a _big_ smell, not of bad architecture or design, but of over-engineering. I'm not arguing this is not SOLID, but IMO this is overkill: are we ever going to need another memory usage provider? The only case I can think of is having a `FixedMemoryUsageProvider` which could be used for some tests of the Infection command, but in this case I would also introduce a `MemoryProvider` interface

Reply:
I see your point. While I don't think it is a smell, I agree we can simplify this particular code.